File: Best Time to Buy and Sell Stock II.java
public class Solution {
    public int maxProfit(int[] prices) {
        // Start typing your Java solution below
        // DO NOT write main() function

        int maxprofit = 0; 
        for(int i = 1; i < prices.length; i++)
        {
            if(prices[i-1]<prices[i])
            {
                maxprofit += prices[i] - prices[i-1];
            }
        }
        
        return maxprofit;
    }
}

File: Best Time to Buy and Sell Stock II.cpp
class Solution {
public:
    int maxProfit(vector<int> &prices) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        int profit = 0;
        int N = prices.size();
        
        for(int i = 1; i < N; i++)
        {
            if(prices[i]>prices[i-1])
            profit +=prices[i]-prices[i-1];
        }
        
        return profit;
    }
};
